canonical-web-and-design/maas-ui
# fix(ui): show correct count when deleting items from list

## Done

- Fixed case where deleting multiple machines/KVMs would show an incorrect count, due to the items being removed from state.
- Driveby - fix action form errors for actions longer than one word e.g "mark-fixed", "override-failed-testing". They needed to be converted to camel case to match the action names, but it wasn't obvious because they are identical for one word actions.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Create a bunch of machines
- Select them all and delete from the action menu
- Check that the processing label updates correctly, i.e. "Deleting 1 of X machines...", "Deleting 2 of X machines...", etc
- Create another machine
- Abort commissioning before it finishes
- Try to mark it as fixed and check that an error shows

## Fixes

Fixes #1770 
Fixes #2175
